mmtk/src/julia_scanning.rs

```diff
@@ -559,6 +569,22 @@ pub fn process_slot<EV: SlotVisitor<JuliaVMSlot>>(closure: &mut EV, slot: Addres
 //     mark_metadata_scanned(obj);
 // }
 
+#[inline(always)]
+pub fn mmtk_decode_restriction_value(pku: u64) -> usize {
+    #[cfg(target_pointer_width = "64")]
+    {
+        // need to apply (pku & ~0x7) to get the object to be traced
+        // so we use (pku & 0x7) to get the offset from the object
+        // and pass it to process_offset_slot
+        return pku as usize & 0x7;
+    }
+
+    #[cfg(not(target_pointer_width = "64"))]
+    {
+        return 0;
```

Comment:
Is this a placeholder value, or the actual value we should use for 32bits?

Reply:
The actual value. I've added a comment to make it clear.